app/services/candidate_interface/courses_recommender.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+module CandidateInterface
+  class CoursesRecommender
+    include Rails.application.routes.url_helpers
+
+    def self.recommended_courses_url(candidate:, locatable: nil)
+      new(candidate:, locatable:).recommended_courses_url
+    end
+
+    # @param [Provider, Site, ApplicationForm] locatable
+    def initialize(candidate:, locatable: nil)
+      @candidate = candidate
+      @locatable = locatable
+    end
+
+    def recommended_courses_url
+      find_url_with_query_params
+    end
+
+  private
+
+    attr_reader :candidate, :locatable
+
+    def find_url_with_query_params
+      uri = URI("#{find_url}results")
+      uri.query = query_parameters.to_query unless query_parameters.empty?
+      uri.to_s
+    end
+
+    def query_parameters
+      params = {}
+
+      params[:can_sponsor_visa] = can_sponsor_visa if can_sponsor_visa
+      # May need to be converted to use the API
+      params[:degree_required] = degree_required if degree_required
+      params[:funding_type] = funding_type if funding_type
+      params[:qualification] = qualification if qualification
+      params[:study_type] = study_type if study_type
+      # May need to be converted to use the API
+      params[:subjects] = subjects if subjects
+      # May need to be converted to use the API
+      params.merge!(location_params)
+
+      params
+    end
+
+    def can_sponsor_visa
+      return unless candidate.application_forms.exists?(personal_details_completed: true)
+
+      # Does the Candidate require Courses to sponsor their visa?
+      # Returns true if the Candidate does not have the right to work or study in the UK
+      requires_visa = candidate.application_forms.exists?(right_to_work_or_study: 'no')
+      requires_visa.to_s # 'true' or 'false'
+    end
+
+    def degree_required
+      return unless candidate.application_forms.exists?(degrees_completed: true)
+
+      candidate_degree_grades = candidate.degree_qualifications.pluck(:grade)
+
+      return 'not_required' if candidate_degree_grades.empty?
+
+      # What Course degree entry requirements can the Candidate meet?
+      return 'show_all_courses' if candidate_degree_grades.include?('First-class honours')
```

Comment:
```suggestion
      # What Course degree entry requirements can the Candidate meet?
```
Since we are going to show_all_courses for everything by a 2:2 and 3rd, this line isn't necessary